client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -192,14 +192,14 @@ public void updateAsync(UpdateRequest updateRequest, ActionListener<UpdateRespon
             listener.onFailure(validationException);
             return;
         }
-        Request req = null;
         try {
-            req = requestConverter.apply(request);
-        } catch (IOException e) {
+            Request req = requestConverter.apply(request);
+
+            ResponseListener responseListener = wrapResponseListener(responseConverter, listener, ignores);
+            client.performRequestAsync(req.method, req.endpoint, req.params, req.entity, responseListener, headers);
+        } catch (Exception e) {
```

Comment:
Yes, that what I did first but then I wrapped everything because I mixed up the performRequestAsync methods.